### ISSUE TITLE
s3 upload speeds too slow for very large files

### ISSUE THREAD

**Issue by gWOLF3** [aws/aws-cli#6541]
Confirm by changing [ ] to [x] below:
- [ x] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [ x] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution

Issue is about usage on:
- [ ] Service API : I want to do X using Y service, what should I do?
- [x ] CLI : passing arguments or cli configurations.
- [ ] Other/Not sure.

**Platform/OS/Hardware/Device**
What are you running the cli on?

Ubuntu 20 LTS

**Describe the question**

AWS Cli `cp` upload is simply not fast enough from instance to s3 bucket in same region for very large file size.

expected file size: over 100GB

aws s3 config:
```
[default]
s3 =
    max_bandwidth = 100GB/s
    multipart_chunksize = 500MB
    multipart_threshold = 5GB
```

My command:
```
aws s3 cp very_large_file.blob s3://mybucket/ --expected-size 100000000000000
```

the problem: 

my upload speed is maxing out only around 63mbps, which takes way too long. both s3 and instance are in the same region. 

what must I be doing wrong?

**Logs/output**
```
Completed 769.5 MiB/101.3 GiB (54.7 MiB/s) with 1 file(s) remaining
```

**Comment by stobrien89** [comment]
Hi @gWOLF3, 

Sorry to hear about the issues.  Assuming the file is a file stream, the expected size in bytes appears to be too large— this parameter is used to calculate the part size (for up to 10,000 parts), so you want to try to get as close to the actual size as possible.  Otherwise, you shouldn't need to include the `--expected-size` option and you may want to try toggling different multipart chunk sizes, or testing with that config var removed altogether.  Let me know either way!

**Comment by gWOLF3** [comment]
Hey @stobrien89 - thanks for your response: 

Your right, the expected size variable I was using was too large (I added 3 too many zeros) - however it appears to not make a difference. I have tried with both using the exact correct `--expected-size` and with that variable not included. 

No matter what I seem to toggle, I cannot seem to get much more than 65MiB/second. This is prohibitively slow for my use case. 

According to the AWS documentation, I should be able to get [up to 100 Gb/second bandwidth between EC2 and S3 endpoints in the same region](https://aws.amazon.com/premiumsupport/knowledge-center/s3-maximum-transfer-speed-ec2/#:~:text=Traffic%20between%20Amazon%20EC2%20and,IPs%20in%20the%20same%20Region.). 

I am going through directly to a VPC endpoint by using the IAM permissions to directly write to S3 by URI. Is there any options here I may be missing? The bandwidth of writes between my collapsable EC2 instances to S3 object storage is absolutely crucial for my application.
